HealthcareApp/src/Healthcare.Application/Appointments/Queries/GetRequestedAppointmentsQueryHandler.cs

```diff
@@ -0,0 +1,34 @@
+using Application.Abstractions;
+using Application.Abstractions.Decorators;
+using Domain.Entities;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace Healthcare.Application.Appointments.Queries;
+
+public class
+    GetRequestedAppointmentsQueryHandler : IRequestHandler<GetRequestedAppointmentsQuery, List<Appointment>>
+{
+    private readonly IAppointmentRepository _appointmentRepository;
+    private readonly IUserManagerDecorator<ApplicationUser> _userManager;
+
+
+    public GetRequestedAppointmentsQueryHandler(IAppointmentRepository appointmentRepository,
+        IUserManagerDecorator<ApplicationUser> userManager)
+    {
+        _appointmentRepository = appointmentRepository;
+        _userManager = userManager;
+    }
+
+    public async Task<List<Appointment>> Handle(GetRequestedAppointmentsQuery request,
+        CancellationToken cancellationToken)
+    {
+        var users = await _userManager.GetAllUsersAsync();
+
+        var doctor = await users.FirstOrDefaultAsync(x => x.Id == request.DoctorId, cancellationToken);
```

Comment:
Refers to all your FirstOrDefault calls. Id comes from identification, something unique, so when you search by id you better use `SingleOrDefault`, because you don't expect more than one record to be returned. Please read the difference